core/src/main/java/org/infinispan/interceptors/locking/AbstractLockingInterceptor.java

```diff
@@ -102,9 +105,24 @@ public Object visitRemoveCommand(InvocationContext ctx, RemoveCommand command) t
       return visitDataWriteCommand(ctx, command);
    }
 
+   Object handleRemoveExpiredException(InvocationContext ctx, RemoveExpiredCommand removeExpiredCommand, Throwable throwable) throws Throwable {
```

Comment:
Wouldn't be better to handle the exception in `InvocationContextInterceptor`? 
If you ignore the exception here, the interceptors before this one will never see the exception.

Reply:
Sorry, maybe I misunderstood the `andExceptionally` method but why would the prior interceptors on the stack not see the exception? It should be caught and converted to an exception stage I thought, no?

Reply:
I mean, they wouldn't see the `TimeoutException`. You are returning `null` in the branch.

Reply:
Yeah, that is the goal for this case. We don't want the TimeoutException to propagate when it has the given flag. This is what is used by expiration removal, unless you mean there is a different case that this shouldn't happen?

Reply:
right :)

Reply:
my initial thought was right. 
Here (https://github.com/infinispan/infinispan/blob/master/core/src/main/java/org/infinispan/expiration/impl/ClusterExpirationManager.java#L360) you are retrying the command if you get a `TimeoutException`. 
This code is never executed now (and it explains the failures in CI).

You need to code this in the `InvocationContextInterceptor` since you want to avoid the logging and not swallow the exception. Something like this: https://github.com/infinispan/infinispan/blob/master/core/src/main/java/org/infinispan/interceptors/impl/InvocationContextInterceptor.java#L120

[or change the tests :rofl: ]

Reply:
Well failures in CI alone is a good enough reason to investigate 🤣

Reply:
But the more I am thinking about it, even putting it at the InvocationContextInterceptor will also break the tests. This needs to be changed in a different way.

Reply:
the example in my comment, the exception is still thrown but it is logged as DEBUG instead of WARN. isn't it what you want?

Reply:
Sure, that is close, I would make it TRACE instead though as well. I believe this also then requires changes on the originator side though as well. All I am saying is this needs to be reevaluated is all and thus why I put the Changes Required.